homeassistant/helpers/config_validation.py

```diff
@@ -1122,7 +1122,10 @@ def raise_issue() -> None:
         # pylint: disable-next=import-outside-toplevel
         from .issue_registry import IssueSeverity, async_create_issue
 
-        with contextlib.suppress(HomeAssistantError):
+        # HomeAssistantError is raised if called from the wrong thread
+        # KeyError is raised if the issue registry is not set up, e.g.
+        # if validation happens from check_config
+        with contextlib.suppress(HomeAssistantError, KeyError):
```

Comment:
I'm not sure if this is the correct change, or if check_config should instead set up the issue_registry?

Reply:
We make sure that all other registries are setup before checking the config in the script here:
https://github.com/home-assistant/core/blob/d6a7127b845e420280971082cbdb655d64ed1c76/homeassistant/scripts/check_config.py#L236-L239

Reply:
Yeah, but at the same time I don't think we want the `check_config` script to create persistent issues, that's why I'm unsure.

Reply:
Maybe patch the issue registry save to storage method, so nothing is saved to storage?

Reply:
Updated by adding a read-only mode to the issue registry